RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+#ifndef DAClusterizerInZT_vect_h
+#define DAClusterizerInZT_vect_h
+
+/**\class DAClusterizerInZT_vect
+
+ Description: separates event tracks into clusters along the beam line
+
+	Version which auto-vectorizes with gcc 4.6 or newer
+
+ */
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+#include <memory>
+#include <stdlib.h>
+#include <malloc.h>
+
+template <typename T, std::size_t N = 16>
+class AlignmentAllocator {
```

Comment:
0) VERY VERY CRITICAL:
please no posix_memalign  nor memalign  at most aligned_alloc that is C and C++ standard and properly implemented by jemalloc itself (so before calling any memory allocator please check that the symbol is in jemalloc.so)



alignment to 16 is not needed, actually not even to 32 as jemalloc seems to return 32-byte aligned memory chuncks (at least to me)

please
1) measure and report, I bet no measureble difference can be observed.
1a) in such a case please remove all code that makes no difference
2) on modern architecture the hardware takes care of alignment independently of which mov instruction the compiler has generated
3) please let a memory allocator do the job of memory allocation
4) if really you want to be forward looking please account for AVX512 as well and make the code able to detect the target architecture (in any case I think this should be a job for the memory allocator not the user code)